src/local_scheduler/local_scheduler_algorithm.cc

```diff
@@ -351,21 +356,18 @@ void add_task_to_actor_queue(LocalSchedulerState *state,
   CHECK(task_counter >= entry->task_counter);
 
   /* Create a new task queue entry. */
-  task_queue_entry *elt = (task_queue_entry *) malloc(sizeof(task_queue_entry));
-  elt->spec = (TaskSpec *) malloc(task_spec_size);
-  memcpy(elt->spec, spec, task_spec_size);
-  elt->task_spec_size = task_spec_size;
+  TaskQueueEntry elt = TaskQueueEntry_init(spec, task_spec_size);
   /* Add the task spec to the actor's task queue in a manner that preserves the
    * order of the actor task counters. Iterate from the beginning of the queue
    * to find the right place to insert the task queue entry. TODO(pcm): This
    * makes submitting multiple actor tasks take quadratic time, which needs to
    * be optimized. */
-  task_queue_entry *current_entry = entry->task_queue;
-  while (current_entry != NULL && current_entry->next != NULL &&
-         task_counter > TaskSpec_actor_counter(current_entry->spec)) {
-    current_entry = current_entry->next;
+  auto it = entry->task_queue->begin();
+  while (it != entry->task_queue->end() &&
+         (task_counter > TaskSpec_actor_counter(it->spec))) {
+    ++it;
   }
-  DL_APPEND_ELEM(entry->task_queue, current_entry, elt);
+  entry->task_queue->insert(it, elt);
```

Comment:
This is a change in semantics (`insert` prepends whereas before we were appending). It looks like the previous approach was incorrect, although maybe the problem never arose because actor methods always arrived in order.